plugins/amdgpu: fix cross compilation

rst0git opened the pull request:
This pull request fixes a problem with cross compilation for the amdgpu plugin (similar to https://github.com/checkpoint-restore/criu/pull/2428) and enables CI testing.

In addition, it allows building RPM/DEB packages that provide the amdgpu-plugin for CPU architectures currently supported by CRIU [^1].

[^1]: https://src.fedoraproject.org/rpms/criu/pull-request/11

avagin commented on plugins/amdgpu/amdgpu_plugin_util.c:
```diff
@@ -193,10 +193,10 @@ void print_kfd_bo_stat(int bo_cnt, struct kfd_criu_bo_bucket *bo_list)
-		pr_info("%s(), %d. KFD BO Size: %llx \n", __func__, idx, bo->size);
-		pr_info("%s(), %d. KFD BO Offset: %llx \n", __func__, idx, bo->offset);
-		pr_info("%s(), %d. KFD BO Restored Offset: %llx \n", __func__, idx, bo->restored_offset);
+		pr_info("%s(), %d. KFD BO Addr: %llx \n", __func__, idx, (unsigned long long)bo->addr);
```
should we use PRIx64 in this case?

rst0git replied:
`PRIx64` seems to be equivalent to `%lx` on the aarch64, ppc64 and mips64el architectures.
```
amdgpu_plugin_util.h:24:20: error: format '%lx' expects argument of type 'long unsigned int', but argument 5 has type 'long long unsigned int' [-Werror=format=]
   24 | #define LOG_PREFIX "amdgpu_plugin: "
      |                    ^~~~~~~~~~~~~~~~~
../../criu/include/log.h:47:52: note: in expansion of macro 'LOG_PREFIX'
   47 | #define pr_info(fmt, ...) print_on_level(LOG_INFO, LOG_PREFIX fmt, ##__VA_ARGS__)
      |                                                    ^~~~~~~~~~
amdgpu_plugin_util.c:198:17: note: in expansion of macro 'pr_info'
  198 |                 pr_info("%s(), %d. KFD BO Offset: %" PRIx64 "\n", __func__, idx, (unsigned long long)bo->offset);
```

avagin replied:
because it has to be:
pr_info("%s(), %d. KFD BO Offset: %" PRIx64 "\n", __func__, idx, bo->offset);

rst0git replied:
```
amdgpu_plugin_util.h:24:20: error: format '%lx' expects argument of type 'long unsigned int', but argument 5 has type '__u64' {aka 'long long unsigned int'} [-Werror=format=]
   24 | #define LOG_PREFIX "amdgpu_plugin: "
      |                    ^~~~~~~~~~~~~~~~~
../../criu/include/log.h:47:52: note: in expansion of macro 'LOG_PREFIX'
   47 | #define pr_info(fmt, ...) print_on_level(LOG_INFO, LOG_PREFIX fmt, ##__VA_ARGS__)
      |                                                    ^~~~~~~~~~
amdgpu_plugin_util.c:198:17: note: in expansion of macro 'pr_info'
  198 |                 pr_info("%s(), %d. KFD BO Offset: %" PRIx64 "\n", __func__, idx, bo->offset);
      |                 ^~~~~~~
```
https://github.com/rst0git/criu/actions/runs/10019074745/job/27695225397

avagin replied:
https://github.com/avagin/criu/commit/cae703bae41d17e3a529e1c4d5d3d1858df60042

avagin replied:
@rst0git this is the trial change, I think we can make it.

rst0git replied:
I have updated the pull request to include the changes from https://github.com/avagin/criu/commit/cae703bae41d17e3a529e1c4d5d3d1858df60042. I can confirm that the checkpoint/restore functionality works as expected with these changes on RHEL 9.4 (kernel version `5.14.0-427.28.1.el9_4.x86_64`) with the Radeon RX 7900 XTX.

rst0git commented:
https://github.com/checkpoint-restore/criu/pull/2452 fixes the following error for cross-compilation with armv7
```
#11 15.87 criu/timer.c:49:17: error: format '%ld' expects argument of type 'long int', but argument 7 has type '__suseconds64_t' {aka 'long long int'} [-Werror=format=]
#11 15.87    49 |         pr_info("Restored %s timer to %" PRId64 ".%ld -> %" PRId64 ".%ld\n", n,
#11 15.87       |                 ^~~~~~~~~~~~~~~~~~~~~~~~
#11 15.87    50 |                 (int64_t)val->it_value.tv_sec, val->it_value.tv_usec,
#11 15.87    51 |                 (int64_t)val->it_interval.tv_sec, val->it_interval.tv_usec);
#11 15.87       |                                                   ~~~~~~~~~~~~~~~~~~~~~~~~
#11 15.87       |                                                                   |
#11 15.87       |                                                                   __suseconds64_t {aka long long int}
```

avagin commented:
LGTM

> clang-format: Possible coding style problem (https://github.com/checkpoint-restore/criu/blob/criu-dev/CONTRIBUTING.md#automatic-tools-to-fix-coding-style)

pls take a look at these warnings and fix them if they are valid.

rst0git commented:
> pls take a look at these warnings and fix them if they are valid.

Thank you for pointing this out! The warnings were valid. I have fixed them and updated the pull request.
